src/file-base.js

```diff
@@ -192,44 +199,55 @@ export class File {
  * @param {number} fileSize Total size of the file
  * @param {string} algorithm sha256/md5 hashing algorithm to use
  * @param {func} progress Callback function with progress
+ * @param {object} file File object
  */
-export function computeHash(fileStream, fileSize, algorithm, progress) {
+export function computeHash(fileStream, fileSize, algorithm, progress, file) {
   return new Promise((resolve, reject) => {
-    let hash = crypto.createHash(algorithm)
-    let offset = 0
-    let totalChunkSize = 0
-    let chunkCount = 0
+    if (file != null && algorithm in file._computedHashes) {
+      if (typeof progress === 'function') {
+        progress(100)
+      }
+      resolve(file._computedHashes[algorithm])
```

Comment:
I think it would be better if we check this in the class level i.e. inside the `hash()` function. So let the class manage this.

Reply:
Makes sense. Looks cleaner as well